### PR TITLE
Feral preppers, survivalists, and militia

### PR DESCRIPTION
**Pull request by BioHazardSuit**
#### Summary

Content "Adds firearm wielding feral humans to town areas; Continuation of https://github.com/CleverRaven/Cataclysm-DDA/pull/51825."

#### Purpose of change

Adds firearm wielding feral humans in form of preppers, survivalists, and militias; Feral versions of them would be mostly left isolated in their safehouses during the chaotic events before the start of the game, making them able to appear and become enemies for the player.

#### Describe the solution

Added 3 feral enemies:
- feral prepper: mainly as a horde bringer. Uses a 9mm pistol with 0 pistol skill, will mostly not miss when beside player. Melee similar to crowbar feral human. 3 amount of bullets available and follows a similar mon-drop of the zombie survivor, focusing on the survival equipment. Resurrects as a survivor zombie.
- crazed survivalist: *Changed to melee. Health follows tough zombie and melee skill follows feral axe human. Follows similar item drop to feral prepper, but more increased in frequency. Being a survivalist, it will mimic the fallen survivor behavior from L4D2 where they flee when (critically) hurt. (I have decided to not make them have heavy survivor armor). *Given a flashlight to introduce surface flashlight enemy and prepare for mad militia.
- mad militia: feral version of the car militia profession. Moderate amount of bullet protection but with weak points similar to the zombie sec-guard (only using a kevlar vest). Uses an automatic gun (ar15) as a weapon and has 8 bullets. Resurrects as soldier zombie, item drop similar to the others; Also uses a headlamp (surface level flashlight enemy/mid-endgame night raid obstacle).

Added 2 (3) locations:
- 2 variations of the 4 bed Lmoe shelters, heavily-moderately(?) trapped and occupied (the idea is that the increasingly feral preppers made more aggressive, lethal deterrents against intruders before becoming mad).
- A variation of a cabin where it is boarded, occupied, and has a basement that leads to a similar, modified Lmoe shelter of the homemade-version.

#### Describe alternatives you've considered

We could add feral security guards (9mm) to some parts of towns, with a `"starts": "# days"` for early-game balance.

#### Testing

Spawning
- Made debug character
- walked around a town, no feral preppers.
- debug to 5th day.
- None appear (quite new to the weight unit in spawning, is it similar to frequency?)
- try experimenting with weight, cost multiplier, and pack size.
- Theory culprit is that the default 60 day start cancels the `“start”: “5 days”` line.
- Try removing the start time code.
- Now they appear.
- Looks good.
- Tried to see if ‘upgrades into monster’ works and debugged time.
- Doesn’t seem to work, whether decreasing half-life.
- Make upgrade_group, since other monsters’ evolutions work with upgrade groups
- Next summer start, mad militias and survivalists appear

Health and mon-drops
- security guard prof
- use gun
- decrease armor due to kill test.
- Drop check.
- loaded guns can drop; militia change to milbase personnel drops for no helmet.
- similar group to survivor zombie, Item drops look alright.

Lethality and behavior
- Aiming check: Prepper misses, mostly hit shot when beside. Survivalists hit shots when they are in the middle-near distance. Militia seems to mostly hit their shots in all ranges, sounds alright.
- Survivalist flees when in the last bar of health. Militia flees at the critical last bar, will mostly die.
- Melee/range lethality: Severely decreased prepper’s melee. Survivalist is alright. Militia firearms are quite lethal.
- Luminance check for militia at night. Works alright.
- ----
2nd Test
- `"Starts":` test
- debugged character Next Summer Start for results.
- Mad militia and survivalists do appear.
- Nice
------
3rd Test
- location test
- locate an occupied lmoe shelter
- enemies do appear, nice
- body controlled a soldier npc through debug (that option is awesome) and went in and fought.
- fought resulted in a moderate injured bleeding arm.
- Sounds dangerous enough.

Survivalist melee test
- Made survivalist fight a crossbow npc.
- Result: Survivalist ends up fleeing, but npc retains heavy bleeding injured in 2 limbs, and 2 other broken limbs.
- sounds maybe alright.

#### Additional context

Screenshot (82)
Occupied shelter
Screenshot (81)
Another occupied shelter but in 2 years.
Screenshot (84)
Fight result
Screenshot (87)
Cabin prepper surface
Screenshot (88)
Cabin prepper shelter basement

### CODE REVIEW

**Comment by BioHazardSuit** [comment]
> > Doesn't the evolution here break the lore a little? Either ferals are improving their skills and equipment over time (shouldn't be smart enough for that?), or new people are going feral long after the initial wave.
> 
> I think this has more to do with gamification that we are trying to avoid gameover situations before the player has the ability to prepare for it. And one of the few ways we have of doing this is via delayed appearance.

I have/will put a `"//":` note about the early game balancing in the feral upgrade group

**Comment by Amoebka** [comment]
What I'm saying is there is a difference between delayed spawns and evolutions. Veteran survivor zombies spawn much later than normal survivor zombies, in the same monstergroups, but the latter don't mutate into the former. The reason for that was explicitly that zombies can't acquire more equipment after death.

IMO, ferals shouldn't evolve at all. The later game variants should be added to the same groups, and the spawn delays handled via the "ends" and "starts" properties. Again, see survivor zombies for reference. Would be nice to keep this stuff consistent.

**Comment by BioHazardSuit** [comment]
> What I'm saying is there is a difference between delayed spawns and evolutions. Veteran survivor zombies spawn much later than normal survivor zombies, in the same monstergroups, but the latter don't mutate into the former. The reason for that was explicitly that zombies can't acquire more equipment after death.
> 
> IMO, ferals shouldn't evolve at all. The later game variants should be added to the same groups, and the spawn delays handled via the "ends" and "starts" properties. Again, see survivor zombies for reference. Would be nice to keep this stuff consistent.

Ah alright, gotta stay consistent. I'll follow the veteran zombie `"starts":`, delete the upgrade group too then. Thank you.

**Comment by BioHazardSuit** [comment]
Alright, everything looks good. Now we wait for a workflow checkup.

**Comment by BioHazardSuit** [comment]
Alright, now everything looks great. Now we wait for a workflow checkup.

**Comment by BioHazardSuit** [comment]
Everything should be alright now.

**Comment by BioHazardSuit** [comment]
Ok, surely now I have made all the changes and improvements I could think of, and that should be all. now I wait for further instructions or suggestions from Devs.

**Comment by BioHazardSuit** [comment]
Ok, maybe, the General build matrix could pass the test now.

**Comment by BioHazardSuit** [comment]
Now I wait for further instructions from Devs.

**Comment by BioHazardSuit** [comment]
ok, file conflict is resolved.

**Comment by BioHazardSuit** [comment]
After some thinking, I'm gonna study how cabins and LMOE's work, and make a booby-trapped, feral inhabited versions of them; only the enemies is kinda half-full, so trying to make other prepper locations for them right now sounds great.
The mapgen, palletes and stuff looks intimidating, but I'm slowly getting it.

**Comment by BioHazardSuit** [comment]
Now I wait for further instructions.

**Comment by Maleclypse** [comment]
If I haven’t reviewed this in the next few days please ping me.

**Comment by BioHazardSuit** [comment]
Alright
I try to fix general build matrix now.

**Comment by BioHazardSuit** [comment]
>  click on the "Details" link, and read every single word and link on that page until you find out why your test is failing. Then fix it and push a new set of changes.

I should read all of it first.
5 days sounds like a alright amount, I will try to ping for review; @Maleclypse

**Comment by I-am-Erk** [comment]
There are a lot of general changes to monster groups going on here besides just adding these guys, what's up with that?

**Comment by BioHazardSuit** [comment]
> There are a lot of general changes to monster groups going on here besides just adding these guys, what's up with that?

Oh right. There were if I remember correctly, 2 other pull requests: one that removed the mon_null on the top of most monster groups, and the other that made changes for the monsters that spawn with the `pack_size` command. Therefore I then had an outdated copy, so I tried to follow the changes here. 
It's been a while amount of time, so I may have forgotten the accurate context.

**Comment by BioHazardSuit** [comment]
> `06:34:29.625 ERROR : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\map.cpp:369 [detach_vehicle] detach_vehicle can't find it!  name=custom_empty, submap:0,0,0
[ Previous repeated 94 times ]
06:34:30.202 ERROR : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\map.cpp:369 [detach_vehicle] detach_vehicle can't find it!  name=custom_empty, submap:0,0,0`

 I need help, what should I do?

**Comment by NetSysFire** [comment]
Nothing. This is a known issue with the tests. Every PR has this error currently, so do not worry.

**Comment by BioHazardSuit** [comment]
> Nothing. This is a known issue with the tests. Every PR has this error currently, so do not worry.

Thank you

**Comment by I-am-Erk** [comment]
For the moment, I don't think any feral armed with guns should be in any of the vanilla spawning groups that can just appear anywhere. Having them as special spawns in LMOE shelters and cabins and such is great, and gives a way to test them out. Having them appear on the street and gank you unexpectedly, less so.

**Comment by BioHazardSuit** [comment]
> For the moment, I don't think any feral armed with guns should be in any of the vanilla spawning groups that can just appear anywhere. Having them as special spawns in LMOE shelters and cabins and such is great, and gives a way to test them out. Having them appear on the street and gank you unexpectedly, less so.

Aw alright then, I'll take the gun ones out most of the groups.

File: data/json/monsters/feral_humans.json [hunk 1]
```diff
@@ -58,6 +58,66 @@
       "DROPS_AMMO"
     ]
   },
+  {
+    "id": "mon_feral_sapien_spear",
+    "type": "MONSTER",
+    "name": { "str": "cunning feral" },
+    "description": "In this feral human's eyes is more intelligence than in those of most of their kind.  They're carrying a crude spear, and have assembled some basic armor from scavenged clothing and bits of riot armour, leaving several gaps.  Although a bit more together, this creature is still only a few steps away from being a living zombie themself.",
```
Comment (Night-Pryanik) [inline data/json/monsters/feral_humans.json:65]:
```suggestion
    "description": "In this feral human's eyes is more intelligence than in those of most of their kind.  They're carrying a crude spear, and have assembled some basic armor from scavenged clothing and bits of riot armor, leaving several gaps.  Although a bit more together, this creature is still only a few steps away from being a living zombie themself.",
```

**Pull request merged**